hub/src/main/java/ch/prevo/open/hub/nodes/NodeCaller.java

```diff
@@ -1,21 +1,22 @@
 package ch.prevo.open.hub.nodes;
 
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+
+import java.util.List;
+import javax.inject.Inject;
 
-import ch.prevo.open.encrypted.model.CapitalTransferInformation;
-import ch.prevo.open.encrypted.model.CommencementMatchNotification;
-import ch.prevo.open.encrypted.model.InsurantInformation;
-import ch.prevo.open.encrypted.model.TerminationMatchNotification;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
 
-import javax.inject.Inject;
-import java.util.List;
-
-import static java.util.Arrays.asList;
-import static java.util.Collections.emptyList;
+import ch.prevo.open.encrypted.model.CapitalTransferInformation;
+import ch.prevo.open.encrypted.model.CommencementMatchNotification;
+import ch.prevo.open.encrypted.model.InsurantInformation;
+import ch.prevo.open.encrypted.model.TerminationMatchNotification;
+import ch.prevo.open.hub.repository.NotificationRepository;
 
 @Service
 public class NodeCaller {
```

Comment:
Would try to keep this class as free of non-rest logic as possible, and rather do the NotificationRepository calls in the NodeService

Reply:
What's the advantage of it? The idea behind it, was that the caller doesn't have to be aware of already notified matches. E.g. if someone else would use the `NodeCaller#postCommencementNotification` one needs to check again if this notification was not already sent?